csharp/src/Ice/Communicator-Properties.cs

```diff
@@ -144,6 +144,64 @@ internal PropertyValue(string v, bool u)
             }
         }
 
+        /// <summary>
+        /// Gets the value of a property as a TimeSpan. If the property is not set, returns null.
+        /// The value must be an integer followed immediately by a time unit of 'ms', 's', 'm', or 'h'.
+        /// These correspond to milliseconds, seconds, minutes, and hours, respectively.
+        /// e.g. 50ms, 3m
+        /// </summary>
+        /// <param name="name">The property name.</param>
+        /// <returns>The property value parsed into a TimeSpan or null.</returns>
+        public TimeSpan? GetPropertyAsTimeSpan(string name)
+        {
+            lock (_properties)
+            {
+                if (_properties.TryGetValue(name, out PropertyValue? pv))
+                {
+                    pv.Used = true;
+
+                    // Match an integer followed by ms, s, m, or h
+                    Match match = Regex.Match(pv.Val, @"^(\d+)(ms|s|m|h)$");
```

Comment:
Should we maybe consider allowing these to be case-insensitive, or should we require lowercase?
Pretty trivial either way, just a thought I had.

Reply:
Also I think we should just accept [a-z]+ instead of the specific (ms|s|m|h), since we already check for the suffix and throw an exception for invalid ones in the switch statement. Otherwise the 'default' statement in it will never be used.

Reply:
We should require lowercase for ms, s etc.